masq/src/commands/financials_command.rs

```diff
@@ -58,26 +161,813 @@ impl Command for FinancialsCommand {
+        matches.value_of("top").map(|str| TopRecordsConfig {
+            count: str
+                .parse::<u16>()
+                .expect("top records count not properly validated"),
+            ordered_by: matches
+                .value_of("ordered")
+                .expect("should be required and defaulted")
+                .try_into()
+                .expect("Clap did not catch invalid value"),
+        })
+    }
+
+    fn parse_range_query_arg<'a, T>(
+        matches: &'a ArgMatches,
+        parameter_name: &'a str,
+    ) -> Option<RangeQueryInputs<T>>
+    where
+        T: FromStr<Err = ParseIntError> + TryFrom<i64> + CheckedMul + Display,
+        <T as TryFrom<i64>>::Error: Debug,
+    {
+        let parse_params =
+            |time_range: &[&str], money_range: &str| -> ((u64, u64), (T, T, String, String)) {
+                (
+                    Self::parse_time_params(time_range).expect("blown up after validation"),
+                    Self::parse_masq_range_to_gwei(money_range).expect("blown up after validation"),
+                )
+            };
+
+        matches.value_of(parameter_name).map(|pair_of_ranges| {
+            //this is already after tight validation
+            let mut separated_ranges = pair_of_ranges.split('|');
+            let time_range = separated_ranges
+                .next()
+                .expectv("first range")
+                .split('-')
+                .collect::<Vec<&str>>();
+            let (
+                (min_age, max_age),
+                (min_amount_num, max_amount_num, min_amount_str, max_amount_str),
+            ) = parse_params(&time_range, separated_ranges.next().expectv("second range"));
+            (
+                Some(RangeQuery {
+                    min_age_s: min_age,
+                    max_age_s: max_age,
+                    min_amount_gwei: min_amount_num,
+                    max_amount_gwei: max_amount_num,
+                }),
+                Some((
+                    (time_range[0].to_string(), time_range[1].to_string()),
+                    (min_amount_str, max_amount_str),
+                )),
+            )
+        })
+    }
+
+    fn print_gwei_or_masq_unit_type(gwei: bool) -> &'static str {
+        if gwei {
+            "Gwei"
+        } else {
+            "MASQ"
+        }
+    }
+
+    fn prepare_headings_of_records(is_gwei: bool) -> (HeadingsHolder, HeadingsHolder) {
+        let balance = if !is_gwei {
+            "Balance [MASQ]"
+        } else {
+            "Balance [Gwei]"
+        };
+        (
+            HeadingsHolder {
+                words: vec!["#", "Wallet", "Age [s]", balance, "Pending tx"],
+                is_gwei,
+            },
+            HeadingsHolder {
+                words: vec!["#", "Wallet", "Age [s]", balance],
+                is_gwei,
+            },
+        )
+    }
+
+    fn convert_masq_from_gwei_and_dress_well<T>(gwei: T) -> String
```

Comment:
There might be formatting options for `format!()` that would take care of most of this for you.

doc.rust-lang.org/std/fmt/index.html